src/sentry/spans/consumers/process/flusher.py

```diff
@@ -134,23 +171,32 @@ def produce(payload: KafkaPayload) -> None:
                 else:
                     backpressure_since.value = 0
 
-                healthy_since.value = system_now
+                # Update healthy_since for all shards handled by this process
+                for healthy_since in healthy_since_list:
+                    healthy_since.value = system_now
 
                 if not flushed_segments:
                     time.sleep(1)
                     continue
 
-                with metrics.timer("spans.buffer.flusher.produce"):
-                    for _, flushed_segment in flushed_segments.items():
-                        if not flushed_segment.spans:
-                            continue
-
-                        spans = [span.payload for span in flushed_segment.spans]
-                        kafka_payload = KafkaPayload(None, orjson.dumps({"spans": spans}), [])
-                        metrics.timing("spans.buffer.segment_size_bytes", len(kafka_payload.value))
-                        produce(kafka_payload)
-
-                with metrics.timer("spans.buffer.flusher.wait_produce"):
+                for shard in shards:
+                    with metrics.timer("spans.buffer.flusher.produce", tags={"shard": shard}):
+                        for _, flushed_segment in flushed_segments.items():
```

Comment:
I'm confused by this logic. This looks like you are sending the same spans to every shard?

Reply:
yeah this is nonsense... leftover from a prev version